Add source checks to test.

codebybrett opened the pull request:
Added source checks to core-tests.r so as to add to error count.

Check for malloc.
Check for whitespace at end of line.
Check that function prototype identifier or REBNATIVE parameter matches introductory comment.
Check line lengths.
Failed checks written to source-analysis.log.
Lines exceeding 79 characters do not count toward error count for the moment.

hostilefork commented:
Brilliant!

This choice was definitely the better strategic move for us.  Instead of the analysis code and its libraries becoming a liability for bootstrap, they're an asset by being a stress-test in its own right of the produced executable.  And the same work gets done.

Awesome work, and a path for future expansion.

Merged and submodule committed with the two easiest fixes:

https://github.com/metaeducation/ren-c/commit/a02a60dd90bad8d81670d56d8c60b9e586bd3264

hostilefork commented:
Random new idea on the suggestion board... noticing when there aren't two newlines spacing functions.

    //
    //  Something: C
    //
    void Something(void)
    {
    }

    //
    //  SomethingElse: C
    //
    void SomethingElse(void)
    {
    }

I'm not sure how it factors into the overall rulespace, but if one function follows another I'd personally like to see that done uniformly as two spaces between (if that is easy).

codebybrett commented:
I'll have a look at that.

hostilefork commented:
I was looking into updating the way in which C identifiers are produced from Rebol words, and trying to kind of consolidate some things about that.  So I changed it, and then that failed a test regarding the correspondence of the names of the identifiers to the names of the comments.

This led me to source-tools.reb where there is an *inverse* transform that turns C identifiers to Rebol, via [c-id-to-word](https://github.com/metaeducation/ren-c-test/blob/f5a648ddba2254575afd3202d02920bc1d55fab1/source-tools.reb#L26)

What I had done to get things moving on what I was trying to do was to go ahead and move the `to-c-name` routine into %common.r, and try and spruce it up.  Having done so I don't think we probably need a reverse transform, I adapted source-tools to this and it passed:

    either find/match mold proto-parser/data/2 {native} [
        unless (
            equal?
                 proto-parser/proto.arg.1
                 (to-c-name to word! proto-parser/data/1)
        ) [
            line: line-of text proto-parser/parse.position
            emit [
                id-mismatch
                (mold proto-parser/data/1) (file) (line)
            ]
        ]
    ][....

So that's what I was thinking of going with, and leave the reversible transforms to Haskell for the moment.  :-)

However I ran into an issue that source-tools has tabs.  Don't want to trip up anything you're doing so I figured I'd check w/you first...

codebybrett commented:
No prob. The c-id-to-word was developed in the early days before I
discovered the transform in the orther direction and for expediency I
dropped it in.

You can kill the tabs will not affect me and I know you'll get pleasure
from that. Been using tabs for as long as I've been using Rebol and it will
take me some time to retrain myself. I probably need to change my script
cleaning script to use spaces instead of tabs and maybe see if I can
arrange some pre-commit warning for tabs.

On 28 November 2015 at 11:37, Brian Dickens <notifications@github.com>
wrote:

> I was looking into updating the way in which C identifiers are produced
> from Rebol words, and trying to kind of consolidate some things about that.
> So I changed it, and then that failed a test regarding the correspondence
> of the names of the identifiers to the names of the comments.
>
> This led me to source-tools.reb where there is an *inverse* transform
> that turns C identifiers to Rebol, via c-id-to-word
> <https://github.com/metaeducation/ren-c-test/blob/f5a648ddba2254575afd3202d02920bc1d55fab1/source-tools.reb#L26>
>
> What I had done to get things moving on what I was trying to do was to go
> ahead and move the to-c-name routine into %common.r, and try and spruce
> it up. Having done so I don't think we probably need a reverse transform, I
> adapted source-tools to this and it passed:
>
> either find/match mold proto-parser/data/2 {native} [
>     unless (
>         equal?
>              proto-parser/proto.arg.1
>              (to-c-name to word! proto-parser/data/1)
>     ) [
>         line: line-of text proto-parser/parse.position
>         emit [
>             id-mismatch
>             (mold proto-parser/data/1) (file) (line)
>         ]
>     ]
> ][....
>
> So that's what I was thinking of going with, and leave the reversible
> transforms to Haskell for the moment. :-)
>
> However I ran into an issue that source-tools has tabs. Don't want to trip
> up anything you're doing so I figured I'd check w/you first...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/metaeducation/ren-c-test/pull/2#issuecomment-160229660>
> .
>
